File: 백준/Silver/1920. 수 찾기/수 찾기.py

n = int(input())
n_data = list(map(int,input().split()))
m = int(input())
m_data = list(map(int,input().split()))

n_data.sort()
def binarysearch(l,n,start,end):
    if start>end:
        return 0
    mid = (start+end)//2
    if l == n[mid]:
        return 1
    elif l< n[mid]:
        return binarysearch(l,n,start,mid-1)
    else:
        return binarysearch(l,n,mid+1,end)
    
for l in m_data:
    start=0
    end = len(n_data)-1
    print(binarysearch(l,n_data,start,end))

File: 백준/Silver/1920. 수 찾기/수 찾기.swift
import Foundation
Q_1920()
func Q_1920() {
    let N = Int(readLine()!)!
    let A = readLine()!.split { $0 == " " }.map { Int($0)! }
    let _ = Int(readLine()!)!
    let M = readLine()!.split { $0 == " " }.map { Int($0)! }
    
    let ary = A.sorted()
    for m in M {
        print(binarySearch(ary, N, m))
    }

    func binarySearch(_ ary: [Int], _ n: Int, _ key: Int) -> Int {
        var start = 0
        var end = n
        var mid = 0
        while end - start > 0 {
            mid = (start + end) / 2
            if ary[mid] < key {
                start = mid + 1
            } else if ary[mid] == key {
                return 1
            } else {
                end = mid
            }
        }
        return 0
    }
}
